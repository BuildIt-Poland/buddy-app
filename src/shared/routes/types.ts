import React from 'react';

export type RouteProps = {
  url: string;
  component: React.ComponentType<any>;
};

export type Routes = {
  [key: string]: RouteProps | { [key: string]: RouteProps };
};
