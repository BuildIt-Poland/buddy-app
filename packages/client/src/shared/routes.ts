enum ROUTES {
  BASE = '/',
  LOGIN = '/login',
  FORGOT_PASSWORD = '/forgot-password',
  RESET_PASSWORD = '/reset-password/:token',
  ROUTE_404 = '/404',

  TALENT = '/talent',
  TALENT_SELECT_BUDDY = '/talent/buddies',
  TALENT_SELECT_NEWBIE = '/talent/buddies/:buddyId/newbies',
  TALENT_ADD_TASK = '/talent/newbies/:newbieId/add-task',
  TALENT_TASK_DETAILS = '/talent/newbies/:newbieId/tasks/:taskId',
  TALENT_TASKS_LIST = '/talent/newbies/:newbieId/tasks',
  TALENT_ADD_NEWBIE = '/talent/buddies/:buddyId/add-newbie',
  TALENT_ADD_BUDDY = '/talent/add-buddy',
  TALENT_ADD_TALENT = '/talent/add-talent',
  TALENT_DETAILS = '/talent/details',
  TALENT_BUDDY_DETAILS = '/talent/buddies/:buddyId/details',
  TALENT_NEWBIE_DETAILS = '/talent/newbies/:newbieId/details',

  BUDDY = '/buddy/',
  BUDDY_DETAILS = '/buddy/details',
  BUDDY_ADD_TASK = '/buddy/newbies/:newbieId/add-task',
  BUDDY_TASK_DETAILS = '/buddy/newbies/:newbieId/tasks/:taskId',
  BUDDY_TASKS_LIST = '/buddy/newbies/:newbieId/tasks',
  BUDDY_SELECT_NEWBIE = '/buddy/newbies',
  BUDDY_ADD_NEWBIE = '/buddy/add-newbie',
  BUDDY_NEWBIE_DETAILS = '/buddy/newbies/:newbieId/details',

  NEWBIE = '/newbie',
  NEWBIE_DETAILS = '/newbie/details',
  NEWBIE_TASKS_LIST = '/newbie/tasks',
  NEWBIE_TASK_DETAILS = '/newbie/tasks/:taskId',
  NEWBIE_BUDDY_DETAILS = '/newbie/buddy/:buddyId/details',
}

export { ROUTES };
