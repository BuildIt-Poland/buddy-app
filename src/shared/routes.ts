enum ROUTES {
  BASE = '/',
  LOGIN = '/login',
  ROUTE_404 = '/404',
  BUDDY_ADD_TASK = '/buddy/newbies/:newbieId/add-task',
  BUDDY_TASK_DETAILS = '/buddy/newbies/:newbieId/tasks/:taskId',
  BUDDY_TASKS_LIST = '/buddy/newbies/:newbieId/tasks',
  BUDDY_SELECT_NEWBIE = '/buddy/newbies',
  BUDDY_ADD_NEWBIE = '/buddy/add-newbie',
  BUDDY_NEWBIE_DETAILS = '/buddy/newbies/:newbieId/details',
  NEWBIE_TASK_DETAILS = '/newbie/tasks/:taskId',
  NEWBIE_TASKS_LIST = '/newbie/tasks',
  NEWBIE_BUDDY_DETAILS = '/newbie/buddy/:buddyId/details',
}

export { ROUTES };
