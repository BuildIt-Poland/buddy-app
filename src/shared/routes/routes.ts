enum ROUTES {
  BASE = '/',
  LOGIN = '/login',
  ERROR = '/404',
  BUDDY_ADD_TASK = '/buddy/add-task',
  BUDDY_TASK_DETAILS = '/buddy/tasks/:taskId',
  BUDDY_TASKS_LIST = '/buddy/tasks',
  BUDDY_SELECT_NEWBIE = '/buddy/select-newbie',
  BUDDY_ADD_NEWBIE = '/buddy/add-newbie',
  BUDDY_NEWBIE_DETAILS = '/buddy/newbie/:id/details',
  NEWBIE_TASK_DETAILS = '/newbie/tasks/:taskId',
  NEWBIE_TASKS_LIST = '/newbie/tasks',
}

export { ROUTES };
