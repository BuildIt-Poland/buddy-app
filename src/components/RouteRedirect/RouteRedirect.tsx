import React from 'react';
import { useHistory } from 'react-router-dom';
import { ROUTES } from '../../shared/routes';
import userService from '../../utils/user.service';
import { isBuddy, isNewbie } from '../../utils/user-helper';

const RouteRedirect: React.FC = () => {
  const history = useHistory();
  if (!userService.isAuthenticated) {
    history.push(ROUTES.LOGIN);
  } else {
    const { role } = userService.getUser();
    if (isBuddy(role)) {
      history.push(ROUTES.BUDDY_SELECT_NEWBIE);
    } else if (isNewbie(role)) {
      history.push(ROUTES.NEWBIE_TASKS_LIST);
    }
  }
  return <></>;
};

export default RouteRedirect;
