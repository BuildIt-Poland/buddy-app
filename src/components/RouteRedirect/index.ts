import RouteRedirect from './RouteRedirect';

export default RouteRedirect;
