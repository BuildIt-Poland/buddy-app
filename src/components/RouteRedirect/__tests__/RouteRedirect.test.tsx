import React from 'react';
import { MemoryRouter, Route } from 'react-router';
import { create } from 'react-test-renderer';
import RouteRedirect from '../RouteRedirect';
import { ROUTES } from '../../../shared/routes';

jest.mock('../../../utils/user.service', () => ({
  isAuthenticated: true,
  getUser: () => ({
    role: 'BUDDY',
    id: '1234',
  }),
}));

describe('Component - RouteRedirect', () => {
  test('renders correctly', () => {
    const component = create(
      <MemoryRouter initialEntries={[ROUTES.BASE]}>
        <Route path={ROUTES.BASE} exact>
          <RouteRedirect />
        </Route>
      </MemoryRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
