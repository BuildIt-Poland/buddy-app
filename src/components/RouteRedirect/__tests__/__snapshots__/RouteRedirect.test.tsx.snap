// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - RouteRedirect renders correctly 1`] = `null`;
